abstractions/java/lib/src/main/java/com/microsoft/kiota/RequestOption.java

```diff
@@ -3,4 +3,5 @@
 /** Represents a request option. */
 public interface RequestOption {
 
+    public <T extends RequestOption> Class<T> getType();
```

Comment:
a lot of implementations actually define Class<? extends RequestOption> instead. please align on one or the other